### ISSUE TITLE
[ELE-282] Add dbt_columns artifacts

### ISSUE THREAD

**Issue by vitorweiss** [elementary-data/elementary#471]
**Is your feature request related to a problem? Please describe.**

Since there is already a feature that brings the descriptions fields from tables inside dbt_models table, it will be really great if we can have the descriptions from columns too. It can be used in a dashboard to make a data catalog about data to a team that is too conservative and don't want to add another tool to their stack.

**Describe the solution you'd like**

The solution may be simple, just use the same methodology to get descriptions from tables inside .yml files. In other words, transform the granularity of dbt_models in a model that can be like dbt_columns that bring the same kind of informations.

**Describe alternatives you've considered**

I considered exploring the macros inside elementary project to apply the change localy to my project, but I believe it can be better to add here.

**Additional context**

A brief from descriptions field from dbt_models:
![image](https://user-images.githubusercontent.com/113526587/207459576-0577e3f9-429a-4287-a04a-bef2611bc241.png)

**Would you be willing to contribute this feature?**

I would love to contribute to this feature if I have an opportunity

<sub>[ELE-66](https://linear.app/elementary/issue/ELE-66/add-description-fields-from-columns-to-tables)</sub>

**Comment by Hadarsagiv** [comment]
Thanks @vitorweiss for opening this issue! we've added this task to our backlog. We'll keep you updated once we have news about this issue.

**Comment by vitorweiss** [comment]
Thank you very much!!

**Comment by Maayan-s** [comment]
Hi @vitorweiss, 
Sorry for the late response here. The issue looks super popular!

I'll share why we delayed this - 
We were mostly concerned about the volume of this table, and were afraid it will add long time to runs. 
However, we were recently able to implement a caching solution for the artifacts, so they can be updated only in change. 

Anyway we would be more than happy to add this. 
Are you still open to contribute?

I see that this is how the columns are returnd in a graph node:
```
columns={
  'invocation_id': 
      ColumnInfo(name='invocation_id', description='Primary key of this table.', meta={}, data_type='string', quote=None, tags=[], _extra={}), 
  'run_started_at': ColumnInfo(name='run_started_at', description='Timestamp the invocation was started.', meta={}, data_type='string', quote=None, tags=[], _extra={})
  ....
  }
  ```
  
I suggest that the fields of the `dbt_models_columns` table would be:
```
model_unique_id,
column_name,
description,
data_type,
tags,
meta,
quote
```

What do you think?

**Comment by vitorweiss** [comment]
I believe this is great!

I would love to contribute, I can start searching about in macros of dbt
and find out how to handle this descriptions. I believe the columns that
you pointed is already a great start.

I'll keep in touch when I find out about something.

Regards
<https://www.indicium.tech/>


Vitor Gerber Weiss

Analytics Engineer

***@***.***

(48) 98447-7583

(48) 3371-7293

Rua Orlando Phillippi, 100 - 3º Andar - Sala 304 - Saco Grande,
Florianópolis - SC, 88032-700
<https://www.instagram.com/indiciumtech/>
<https://www.facebook.com/indiciumtech/>
<https://www.linkedin.com/company/indiciumtech>

@indicium.tech

**Comment by Maayan-s** [comment]
Cool! 
Some references to help:

1. You will need to create an empty model, [like here](https://github.com/elementary-data/dbt-data-reliability/blob/master/models/edr/dbt_artifacts/dbt_models.sql).
2. [These are the macros](https://github.com/elementary-data/dbt-data-reliability/blob/master/macros/edr/dbt_artifacts/upload_dbt_models.sql) that load data to `dbt_models`. You will basically be running on the same nodes, but extracting different fields.
3. The columns object is a dict within a dict. You can [see here](https://github.com/elementary-data/dbt-data-reliability/blob/master/macros/edr/dbt_artifacts/upload_run_results.sql) how we handled the `timing` dict and do something similar.

**Comment by vitorweiss** [comment]
Thank you for the help.
<https://www.indicium.tech/>


Vitor Gerber Weiss

Analytics Engineer

***@***.***

(48) 98447-7583

(48) 3371-7293

Rua Orlando Phillippi, 100 - 3º Andar - Sala 304 - Saco Grande,
Florianópolis - SC, 88032-700
<https://www.instagram.com/indiciumtech/>
<https://www.facebook.com/indiciumtech/>
<https://www.linkedin.com/company/indiciumtech>

@indicium.tech

**Comment by Maayan-s** [comment]
Released in dbt package 0.7.8 🎉
